t/40mtls.t

```diff
@@ -0,0 +1,56 @@
+use strict;
+use warnings;
+use File::Temp qw(tempdir);
+use Net::EmptyPort qw(empty_port);
+use Test::More;
+use t::Util;
+
+my $tls_port = empty_port();
+my $tls12_flag = '--tls-max 1.2';
+my $good_client_key_cert = '--key misc/test-ca/demoCA/test_client.key --cert misc/test-ca/demoCA/test_client.crt';
+my $wrong_client_key_cert = '--key examples/h2o/server.key --cert examples/h2o/server.crt';
+my $TLS_RE_OK = qr{hello};
+my $TLS_RE_BAD_CERT = qr{ptls_handshake:299};
+
+sub do_test {
+    my $server = start_server();
+    like run_tls_client("", $good_client_key_cert), $TLS_RE_OK, "mTLS13";
+    #like run_tls_client($tls12_flag, $good_client_key_cert), $TLS_RE_OK, "mTLS12";
+
+    unlike run_tls_client("", ""), $TLS_RE_OK, "mTLS13 no client cert";
```

Comment:
Could we be a bit more specific in checking that the cause of the failure is due to the TLS handshake failing? That'd depend on how curl exits when the handshake fails.

Reply:
There is an error from curl when I use "--tls-max 1.2" to force TLS 1.2, but this option failed in CI testing. The same option works fine on my Ubuntu VM and Mac. Not sure what the difference is regarding curl.

Reply:
updated with curl openssl version detection